### PR TITLE
PNDA-36: Flink history server service creation

### PR DESCRIPTION
**Pull request by GaneshManal**

### CODE REVIEW

File: salt/flink/init.sls [hunk 1]
```diff
@@ -68,3 +70,26 @@ flink-create_flink_logs_directory:
     - name: /var/log/pnda/flink
     - user: {{ pnda_user }}
     - makedirs: True
+
+flink-jobmanager_historyserver_archive_dir_initialize-hdfs:
```
Comment (aswinbh) [inline salt/flink/init.sls]: Keep the name as flink-jobmanager_archive_dir_initialize-hdfs as history server config can be different as well

File: salt/flink/init.sls [hunk 2]
```diff
@@ -68,3 +70,26 @@ flink-create_flink_logs_directory:
...
+  cmd.run:
+    - name: 'sudo -u hdfs hdfs dfs -mkdir -p {{ arhive_dir_hdfs_path }}; sudo -u hdfs hdfs dfs -chmod 777 {{ arhive_dir_hdfs_path }}'
```
Comment (aswinbh) [inline salt/flink/init.sls]: typo on arhive

File: salt/flink/init.sls [hunk 3]
```diff
@@ -9,14 +9,14 @@
...
 
-
 {% set pnda_home = pillar['pnda']['homedir'] %}
 {% set pnda_user  = pillar['pnda']['user'] %}
 {% set flink_real_dir = pnda_home + '/flink-' + flink_version %}
 {% set flink_link_dir = pnda_home + '/flink' %}
 
 {% set namenode = salt['pnda.hadoop_namenode']() %}
-{% set flink_hdfs_work_dir = '/user/' + pnda_user + '/flink/work' %}
+{% set archive_dir = 'completed-jobs' %}
```
Comment (aswinbh) [inline salt/flink/init.sls]: let's configure the completed-jobs directory under /users/pnda/flink/

File: salt/flink/templates/flink-service.service.tpl [hunk 4]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=Flink History-Server Service
+
+[Service]
+Type=forking
+WorkingDirectory={{ install_dir }}
+ExecStart={{ install_dir }}/bin/historyserver.sh start
+Restart=always
+RestartSec=2
+
```
Comment (aswinbh) [inline salt/flink/templates/flink-service.service.tpl]: "/bin/historyserver.sh stop" is missing.

File: salt/flink/templates/flink.conf.tpl [hunk 5]
```diff
@@ -89,19 +89,19 @@ web.port: 8081
...
-#historyserver.web.address: 0.0.0.0
+historyserver.web.address: 0.0.0.0
 
 # The port under which the web-based HistoryServer listens.
-#historyserver.web.port: 8082
+historyserver.web.port: 8082
 
 # Comma separated list of directories to monitor for completed jobs.
-#historyserver.archive.fs.dir: hdfs:///completed-jobs/
+historyserver.archive.fs.dir: {{ namenode }}/{{ path }}
```
Comment (aswinbh) [inline salt/flink/templates/flink.conf.tpl]: put a comment that the value is same as job manager archive dir.

**Review by aswinbh — CHANGES_REQUESTED** [review]

File: salt/flink/init.sls (hunk 1 above)
Comment (GaneshManal) [reply salt/flink/init.sls]: change done.

File: salt/flink/init.sls (hunk 2 above)
Comment (GaneshManal) [reply salt/flink/init.sls]: change done

File: salt/flink/init.sls (hunk 3 above)
Comment (GaneshManal) [reply salt/flink/init.sls]: change done.

File: salt/flink/templates/flink-service.service.tpl (hunk 4 above)
Comment (GaneshManal) [reply salt/flink/templates/flink-service.service.tpl]: added.

File: salt/flink/templates/flink.conf.tpl (hunk 5 above)
Comment (GaneshManal) [reply salt/flink/templates/flink.conf.tpl]: change done.

**Pull request merged**